File: api_wrapper/thecatapi/python/src/models/image.py
import typing

from . import model
from . import breed
from . import category


class Image(model.Model):
    def __init__(self,
                 id: typing.Optional[int] = None,
                 url: typing.Optional[str] = None,
                 width: typing.Optional[int] = None,
                 height: typing.Optional[int] = None,
                 mime_type: typing.Optional[str] = None,
                 breeds: typing.Optional[list[dict]] = None,
                 categories: typing.Optional[list[dict]] = None,
                 breed_ids: typing.Optional[str] = None,
                 **other_fields) -> None:
        super().__init__(id, **other_fields)
        self.url = url
        self.width = width
        self.height = height
        self.mime_type = mime_type
        self.breeds = self.parse(breeds, breed.Breed)
        self.categories = self.parse(categories, category.Category)
        self.breed_ids = breed_ids

    @staticmethod
    def parse(data: list[dict], klass: type[model.Model]) -> list[model.Model]:
        if isinstance(data, dict):
            output = [klass(**data)]
        elif isinstance(data, list):
            output = [klass(**d) for d in data]
        else:
            output = []
        return output

File: api_wrapper/thecatapi/java/src/main/java/model/Image.java
package model;

import java.util.Map;
import java.util.List;

public class Image extends Model {
    private final String url;
    private final String mimeType;
    private final int width;
    private final int height;
    private final List<Breed> breeds;
    private final List<Category> categories;

    public Image(int id, List<Breed> breeds, List<Category> categories, Map<String, Object> data) {
        super(id, data);
        this.url = String.valueOf(data.get("url"));
        this.mimeType = String.valueOf(data.get("mime_type"));
        this.width = (int) data.get("width");
        this.height = (int) data.get("height");
        this.breeds = breeds;
        this.categories = categories;
    }

    public String getUrl() {
        return url;
    }

    public String getMimeType() {
        return mimeType;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public List<Breed> getBreeds() {
        return breeds;
    }

    public List<Category> getCategories() {
        return categories;
    }
}
